### DISCUSSION TITLE
Wrong the rule of thumb in slim mode

### DISCUSSION THREAD

**Discussion by likemusic** [osm2pgsql-dev/osm2pgsql#1433] [General]
According to the [docs](https://osm2pgsql.org/doc/manual.html#caching):
```
The rule of thumb in slim mode is as follows: use the size of the PBF file you are trying to import or about 75% of RAM, whatever is smaller.
```

But in fact seems that cache size must be above twice lager then source file.
I've made tests on 2 pbf-files and it shown that for cache size equal to source file size import speed for ways just 30-40% from max possible.

![image](https://user-images.githubusercontent.com/603401/111852498-3670ee80-8928-11eb-8ff3-4be8090798d7.png)

![image](https://user-images.githubusercontent.com/603401/111852512-44267400-8928-11eb-9033-60eea746a200.png)

[Google sheet with tests results](https://docs.google.com/spreadsheets/d/1IjCndMOyr6cM2FDRu53ASZdIR6ZdmUeqoREf-V9TGGk/edit?usp=sharing).

**Comment by lonvia** [comment]
You haven't explained your test setup and what files you have tested against. So it's hard to comment on this.

**Reply by likemusic** [reply]
In the end of issue message there is a link to google sheet with data source.

There is some explain for columns in this file. Hope this will help:
![image](https://user-images.githubusercontent.com/603401/111866791-06096e80-8981-11eb-9436-0ae0308c178a.png)

**Reply by lonvia** [reply]
These test files are tiny and, yes, for those files the ratio is more 2x PBF because they node cache is completely sparse. The rule in the documentation was written for PBF files in the GB range.

**Reply by likemusic** [reply]
What about add this useful info to the docs?

**Comment by likemusic** [comment]
Also I found that there is a little inconsistent information about --slim mode and required cache size.

In [8.3 Caching](https://osm2pgsql.org/doc/manual.html#caching) section:
```
 The rule of thumb in slim mode is as follows: use the size of the PBF file you are trying to import or about 75% of RAM, whatever is smaller.
```

In [8.5 Middle Command Line Options](https://osm2pgsql.org/doc/manual.html#middle-command-line-options) section:
```
 As a rule of thumb, give a bit more than the size of the import file in PBF format.
```
